File: software/examples/python/example_simple.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change XYZ to the UID of your Solid State Relay Bricklet

import time

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_solid_state_relay import BrickletSolidStateRelay

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    ssr = BrickletSolidStateRelay(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Turn relay on/off 10 times with 1 second delay
    for i in range(5):
        time.sleep(1)
        ssr.set_state(True)
        time.sleep(1)
        ssr.set_state(False)

    raw_input("Press key to exit\n") # Use input() in Python 3
    ipcon.disconnect()

File: software/examples/perl/example_simple.pl
#!/usr/bin/perl

use Tinkerforge::IPConnection;
use Tinkerforge::BrickletSolidStateRelay;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your Solid State Relay Bricklet

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $ssr = Tinkerforge::BrickletSolidStateRelay->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Turn relay on/off 10 times with 1 second delay
for (my $i = 0; $i < 5; $i++)
{
    sleep(1);
    $ssr->set_state(1);
    sleep(1);
    $ssr->set_state(0);
}

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();
